### PR TITLE
Align results of array_first_value/array_last_value with access by index.

### PR DESCRIPTION
**Pull request by vkaverin**
Return `null` for an empty array.

### CODE REVIEW

**Review by unserialize — DISMISSED** [review]

**Comment by unserialize** [comment]
Can you please rename the commit message according to our code style? From a small letter, without a period at the end

**Comment by vkaverin** [comment]
Sure. Pushed.

**Review by unserialize — APPROVED** [review]

File: kphp_polyfills.php [hunk 1]
```diff
@@ -324,9 +324,17 @@ function array_first_key(array $a) {
...
+ * * In KPHP the function always returns result of type `T`.
+ *
  * @ return can't be expressed in phpdoc, done via KPHPStorm plugin for IDE
  */
 function array_first_value(array $a) {
+  if (!$a) {
+    return null;
+  }
   reset($a);
   return current($a);
```
Comment (rybakit) [inline kphp_polyfills.php:339]:
```suggestion
  return $a ? reset($a) : null;
```

File: kphp_polyfills.php [hunk 2]
```diff
@@ -341,9 +349,17 @@ function array_last_key(array $a) {
...
+ * * In KPHP the function always returns result of type `T`.
+ *
  * @ return can't be expressed in phpdoc, done via KPHPStorm plugin for IDE
  */
 function array_last_value(array $a) {
+  if (!$a) {
+    return null;
+  }
   end($a);
   return current($a);
```
Comment (rybakit) [inline kphp_polyfills.php:364]:
```suggestion
  return $a ? end($a) : null;
```

**Review by rybakit — CHANGES_REQUESTED** [review]

**Review by unserialize — APPROVED** [review]

**Pull request merged**